### ISSUE TITLE
Custom template -  issue accessing error key

### ISSUE THREAD

**Issue by renarsvilnis** [blueimp/jQuery-File-Upload#3274]
Hi, im trying to add custom templating engine for uploads and downloads. But i have issue where i can't access the error message for each file in template (trying to upload files that arent in the accepted file type)
Using version 9.7.0 (newest).

I do the fileupload init.

``` javascript
gallery.addClass('fileupload-processing').fileupload({
          url: ubercms.baseUrl,
          maxFileSize: 8 * 1024 * 1024,
          acceptFileTypes: /(\.|\/)(gif|jpe?g|png)$/i,
          uploadTemplateId: null,
          downloadTemplateId: null,
          uploadTemplate: function(o) {
            return uberTemplate.__uploadTemplate(o);
          },
          // downloadTemplate: function (o) {
          //   return 'to-be-added';
          // }
});
```

Using handlebars for this. (not the source of the issue)

``` javascript
__uploadTemplate: function(data) {
      var templateSource = $('#uber-gallery-upload').html(),
          template = Handlebars.compile(templateSource),
          element = $(template(data));

      return element;
}
```

The console after "console.log(data);".
![screen shot 2014-07-31 at 12 01 30](https://cloud.githubusercontent.com/assets/3515099/3761946/4877ff10-1891-11e4-939f-cce935a08a6f.png)

As you see there is error key value in "files" un "files[0]". Which seen weird because "files" is shown is an array of length 1. therefore has only one file with key "0". The problem is that i can't access the "error" key nowhere - not in template and not in "__uploadTemplate"  function through console. I have been following the [wiki guide](https://github.com/blueimp/jQuery-File-Upload/wiki/Template-Engine) for custom template and referencing the example/original template but no luck. I tried going deeper and deeper and outputting the data (code below). Everything returns as expected until "console.log(data.files[0].error);" which returns undefined. Last command "stringify" also doesn't include "error". 

**Later found out that if i set timeout to 0sec, then when trying to output or use the data it will output the correct error. But this isn't a solution for my issue.**
Tried the wiki template which worked, the blueimp template language written template template worked. But it isn't event a handlebar.js problem.

``` javascript
console.log(data); // returns expected results
console.log(data.files); // returns expected results
console.log(data.files[0]); // returns expected results
console.log(data.files[0].error); // returns undefined
console.log(JSON.stringify(data.files[0])); // return doesn't include error

// no timeout
console.log(o.files[0].error); // return undefined

// with 0 timeout
setTimeout(function () {
  console.log(o.files[0].error); // return "File type not allowed"
}, 0);

```

Why is this happening? At the time isn't the object initiated or something?

**UPDATE**
Did some breakpoint testing, seems like the object at first doesn't have error key, but after i build the the template the validation happens after the template has been rendered and the "error" key will be added after the template rendering also.
Got the error message to render by just leaving a empty div with class "error" and the validation add the according text in it. But i the start-upload button for these elements are still visible which i don't want to be visible for files that don't pass the validation for size, type, ...

**UPDATE #2**
After doing breakpoints and inspecting when the data gets the validation. Led me to understand the source code and that the validation happens after the initial template rendering.
Solutions copy-paste "add" function from "jquery.fileupload-ui.js" and ad processing before template rendering and let else deferring functions go as normal after rendering.
Case closed!

**TL;DR**
Follow code through breakpoints and read source code!
